### DISCUSSION TITLE
How to post-process data from Pele calculations?

### DISCUSSION THREAD

**Discussion by RachelMeng0096** [AMReX-Combustion/PeleC#746] [Q&A] [answered]
Recently, I've been attempting to post-process data obtained from Pele simulations. My preferred post-processing software is MATLAB, but I've found it challenging to import Pele's output file formats into MATLAB for analysis. Could someone please advise me on how to achieve this functionality? Or perhaps share how others typically perform post-processing? Would it be possible to provide a sample post-processing script?
I appreciate your help with this matter.

**Comment by marchdf** [comment, marked as answer]
There are many ways one can go about doing this. 
1. PeleAnalysis (https://github.com/AMReX-Combustion/PeleAnalysis): fast postprocessing C++ code for analysing Pele data, with a bunch of neat functionality
2. `yt` (https://yt-project.org): convenient python package for postprocessing Pele (and other) data. Easy to use and full of features and examples. Apparently there's a way to export to matlab friendly things from `yt`. A search for `yt` in the PeleAnalysis repo will show a bunch of examples (https://github.com/search?q=repo%3AAMReX-Combustion%2FPeleAnalysis%20yt&type=code). Other examples can be found here: https://github.com/AMReX-Combustion/PelePlot. And also by searching for `import yt` in the PeleC repo: https://github.com/search?q=repo%3AAMReX-Combustion%2FPeleC+%22import+yt%22&type=code
3. `pyamrex` (http://pyamrex.readthedocs.io): convenient python wrapper of amrex, fast, but need to code your own post-processing routines

As you can see, we like C++ and python ;)

**Comment by drummerdoc** [comment]
And if you're exceptionally talented with VisIt or ParaView, they do have some facilities (slicing, path lines, contours, etc) that could useful - although sometimes there are mysterious interpolations and such behind the scenes you have to be careful about.
